### ISSUE TITLE
Entry OnValidationChanged

### ISSUE THREAD

**Issue by fpabl0** [fyne-io/fyne#3181]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Hello all :), I have noticed that the changes in `entry_validation.go` by using `errors.Is(...)`, actually change the way OnValidationChanged works. That is because `errors.Is` actually compares an error to a value, according to https://go.dev/blog/go1.13-errors:
> In the simplest case, the errors.Is function behaves like a comparison to a sentinel error

Then validators that does not return a pre-existing sentinel error (using for example `return errors.New(...)`) will trigger validation change even if the error has actually the same error message, because they will actually have different values (different instances).

So for example these validators:
```go
entry.Validator = func (s string) error {
     if s == "" {
         return errors.New("Required")
     }
     return nil
}


entry.Validator = func (s string) error {
     _, err := mail.ParseAddress(s)
     return err
}
```

Will trigger OnValidationChange (and also refreshing the widget) even if the error in "practical terms" has not change.

Is this the desired behavior? Note that before that change, the behavior was different and above validators worked as expected, but not sure if this behavior change was intentional.

### How to reproduce

1. Run the example code.
2. Focus the entry and type "abcd".
3. Notice that the error "mail: missing '@' or angle-addr" has been printed 4 times.
4. That error is printed from the OnValidationChanged, normally it should be printed just 1 time, as it is the "same" error.
5. If you run the same code using `fyne.io/fyne/v2@v2.1.4`, that error will print just once.

### Example code

```go
package main

import (
	"fmt"
	"net/mail"

	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Hello")

	e := widget.NewEntry()
	e.Validator = func(s string) error {
		_, err := mail.ParseAddress(s)
		return err
	}
	e.SetOnValidationChanged(func(err error) {
		fmt.Println(err)
	})
	w.SetContent(container.NewVBox(e))

	w.ShowAndRun()
}
```

### Fyne version

6bf574b4a8675676e34b311b115e58e1b50ff824

### Go compiler version

1.18.4

### Operating system

macOS

### Operating system version

Big Sur 11.5.1

**Comment by Jacalz** [comment]
That’s a good question. The current behaviour is technically more in line with what the Go developers suggestions to have a global error variable and then just returning the same value (instead of creating new errors using `errors.New` each time). However, far from everyone uses that approach so you are right that it might not always make sense.

Not using `errors.Is` removes the support for correctly identifying wrapped errors but I suspect that it doesn’t matter much. They are not that common and probably don’t make sense in this context either?

**Comment by fpabl0** [comment]
> The current behaviour is technically more in line with what the Go developers suggestions to have a global error variable and then just returning the same value

Yes, indeed that makes sense. I mean if this behavior is desired, then I think it should be documented as it could cause breaking changes in some rare situations (indeed I've got one in an application 😅).

Another problem is that although using a global error variable is definitely better than created a new one each time, this is just one way of returning error, indeed inside Go source code we found both ways, for example in the package `net/mail` the function `mail.ParseAddress` calls `parseSingleAddress` that is implemented in this way (without sentinel errors):
```go
func (p *addrParser) parseSingleAddress() (*Address, error) {
	addrs, err := p.parseAddress(true)
	if err != nil {
		return nil, err
	}
	if !p.skipCFWS() {
		return nil, errors.New("mail: misformatted parenthetical comment")
	}
	if !p.empty() {
		return nil, fmt.Errorf("mail: expected single address, got %q", p.s)
	}
	if len(addrs) == 0 {
		return nil, errors.New("mail: empty group")
	}
	if len(addrs) > 1 {
		return nil, errors.New("mail: group with multiple addresses")
	}
	return addrs[0], nil
}
```
